google-cloud-core/src/test/java/com/google/cloud/PolicyV3Test.java

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import com.google.cloud.Policy.DefaultMarshaller;
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import org.junit.Test;
+
+public class PolicyV3Test {
+
+  private static final String ALL_USERS = "allUsers";
+  private static final String ALL_AUTH_USERS = "allAuthenticatedUsers";
+  private static final String USER = "user:abc@gmail.com";
+  private static final String SERVICE_ACCOUNT = "serviceAccount:service-account@gmail.com";
+  private static final String GROUP = "group:group@gmail.com";
+  private static final String DOMAIN = "domain:google.com";
+  private static final String VIEWER = "roles/viewer";
+  private static final String EDITOR = "roles/editor";
+  private static final String OWNER = "roles/owner";
+  private static final List<String> MEMBERS_LIST_1 =
+      ImmutableList.of(USER, SERVICE_ACCOUNT, ALL_USERS);
+  private static final List<String> MEMBERS_LIST_2 =
+      ImmutableList.of(ALL_AUTH_USERS, GROUP, DOMAIN);
+  private static final List<Binding> BINDINGS_NO_CONDITIONS =
+      ImmutableList.of(
+          Binding.newBuilder().setRole(VIEWER).setMembers(MEMBERS_LIST_1).build(),
+          Binding.newBuilder().setRole(EDITOR).setMembers(MEMBERS_LIST_2).build());
+  private static final List<Binding> BINDINGS_WITH_CONDITIONS =
+      ImmutableList.copyOf(BINDINGS_NO_CONDITIONS)
+          .of(
```

Comment:
You're calling a static method on an instance, and throwing the instance away. That's not what you want to do. (If the project used Error-Prone, this would have generated a [warning](https://errorprone.info/bugpattern/StaticQualifiedUsingExpression).)

It looks like you want a variant of `BINDINGS_NO_CONDITIONS` where the viewer role has a condition. What about this?

```
ImmutableList.of(
    BINDINGS_NO_CONDITIONS.get(0).toBuilder().setCondition(...).build(),
    BINDINGS_NO_CONDITIONS.get(1))
```

Although it might be nice to have a constant for each element of `BINDINGS_NO_CONDITIONS` (`VIEWER` and `EDITOR`)?